app/values/resource_identifier/solr_repository.rb

```diff
@@ -0,0 +1,62 @@
+class ResourceIdentifier
+  ##
+  # Repository for querying solr for a single or multiple IDs and getting back a
+  # document object which can return its dependants and timestamps.
+  class SolrRepository
+    class << self
+      # @param [Array<Integer> OR Integer] id ID or array of IDs of records to
+      #   query for.
+      # @return [Record, #timestamp, #dependant_ids] Solr-record powered record
+      #   which can return its timestamp and dependant ids.
+      def find(id)
+        id = Array.wrap(id)
+        return CompositeRecord.new({}) if id.blank?
+        records = solr.query("id:(#{id.join(' OR ')})").map do |r|
```

Comment:
Wouldn't the A get reindexed if one of the members is removed?  I'm going to test that out, because it's something we need to account for one way or the other.